compiler/src/main/java/com/dylibso/chicory/compiler/internal/Shaded.java

```diff
@@ -169,110 +169,86 @@ public static void memoryWriteDouble(int base, double value, int offset, Memory
 
     public static int memoryAtomicIntByteRead(int base, int offset, Memory memory) {
         var ptr = getAddr(base, offset);
-        synchronized (memory.lock(ptr)) {
-            return memory.read(ptr);
-        }
+        return Byte.toUnsignedInt(memory.atomicReadByte(ptr));
```

Comment:
Let me reason out loudly:
- assume we release those changes in a 1.7.0 release
- a user compiles using the build time compiler with 1.7.0
- attempting to use the generated classes with 1.6.0 of the runtime is going to break

correct?

Reply:
Yes, calling new runtime methods in `Shaded` will break on an older runtime... what's the rational behind this requirement? Setups where one dynamically pushes newly compiled classes into an otherwise static installation? I don't see any way around it. What does that mean? This PR would require a new major?